packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -127,6 +169,31 @@ export interface DecoratorInfo {
   // A dict of the decorator's arguments. For example, `{ encoding: "base64url" }`.
   arguments: Record<string, any>;
 }
+export interface SdkClientType<TServiceOperation extends SdkServiceOperation>
+  extends DecoratedType {
+  __raw: SdkClient | SdkOperationGroup;
+  kind: "client";
+  name: string;
+  clientNamespace: string; // fully qualified namespace
+  doc?: string;
+  summary?: string;
+  /**
+   * @deprecated Use `init.paramters` instead.
+   */
+  initialization: SdkInitializationType;
+  init: SdkClientInitializationType;
```

Comment:
nit, personally I'd like the full word `initialization`. But appears here we name it `init` because we need to handle the deprecate procedure.

Reply:
i'm not 100% like current name. but not have more idea. @iscai-msft do you have any suggestion. this pr will not break anything, but just deprecate things, so we could not use previous property name.

Reply:
we could change the whole thing to "constructor" instead of "initialization"? Don't know if it's that much work though, I'm ok with `init` if we want to keep it that way

Reply:
We could use `clientInitialization`

Reply:
i switched it to `clientInitialization`